lua/gatekeeper/staticlib.lua

```diff
@@ -74,6 +82,48 @@ function alloc_an_lcore (numa_table)
 	return lcore_t[1]
 end
 
+function merge_arrays (a, b)
+	for i, v in ipairs(b) do
+		table.insert(a, v)
+	end
+	return a
+end
+
+function alloc_lcores_evenly_from_all_numa_nodes (numa_table, n,
+		fixed_lcores_per_numa)
+	local num_numa_nodes = get_num_numa_nodes (numa_table)
+	local q = n / num_numa_nodes
+	local r = n % num_numa_nodes
+	local i = 0
+	local res = {}
+	for numa, lcores in all_ipairs(numa_table) do
+		local lcores_needed = q + ((i < r) and 1 or 0)
+		if lcores_needed > 0 then
+			lcores_needed = lcores_needed + fixed_lcores_per_numa
+		else
+			break
+		end
+		if #lcores >= lcores_needed then
+			local a1, a2 = split_array(lcores, lcores_needed)
+			numa_table[numa] = a2
+			numa_table["__net_conf"].numa_used[numa] = true
```

Comment:
We are repeating these three lines from `alloc_lcores_from_same_numa()`. It's better to define the following function and use both times:
```Lua
function alloc_lcores_at_numa (numa_table, numa, n)
	local a1, a2 = split_array(numa_table[numa], n)
	numa_table[numa] = a2
	numa_table["__net_conf"].numa_used[numa] = true
	return a1
end
```